eclipse/che
# Do not mount ssh keys as secrets if the name is not a valid Kubernetes config map key name

### What does this PR do?
Do not mount ssh keys as secrets if the name is not a valid Kubernetes config map key name

### Screenshot/screencast of this PR



### What issues does this PR fix or reference?
Fixes https://github.com/eclipse/che/issues/19024


### How to test this PR?
-  deploy nightly. 
- create invalid ssh key
```bash
curl $CHE_SERVER_URL/api/ssh \
-H 'Authorization: Bearer '${KEYCLOAK_TOKEN} \
-H "Accept: application/json" \
-H "Content-Type:application/json" \
--data @<(cat <<EOF
  {
    "name": "http://github.com",
    "service": "vcs",
    "publicKey": "ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABAQCFs7ZasKC9XyaX02tYbW/8dWNdoce6hv48uK2Hgfqex+rYT+aV5lqCdVVKfzWQcF3gsM1i9Ee6w+Rr4HQ++QZtE4w2huT82uVcfLy/0Axyh6ZMyVTFpchykl8v93hQo1YaTgsUrokPQp/3gLLfcP9PNBOZn071HJK/zMIcyAjfzS8bCxQtMdI7aZd7ohb4ZiiymsFvoOInjDPKokr1E+GdfAIp4o+g6FZAA4cM7hW0EExzXMBtDEtw4sexiSrZWN6uTbFUVxVtoyKCh3Fyaumj6zVXRHCf+TqUGwURDGooO2RS5K0gzbX6uKSVn9P6uEQdb0f/cPurhX2cb1hRZyHH ",
    "privateKey": "-----BEGIN RSA PRIVATE KEY-----\nMIIEogIBAAKCAQEAhbO2WrCgvV8ml9NrWG1v/HVjXaHHuob+PLith4H6nsfq2E/m\nleZagnVVSn81kHBd4LDNYvRHusPka+B0PvkGbROMNobk/NrlXHy8v9AMcoemTMlU\nxaXIcpJfL/d4UKNWGk4LFK6JD0Kf94Cy33D/TzQTmZ9O9RySv8zCHMgI380vGwsU\nLTHSO2mXe6IW+GYosprBb6DiJ4wzyqJK9RPhnXwCKeKPoOhWQAOHDO4VtBBMc1zA\nbQxLcOLHsYkq2Vjerk2xVFcVbaMigodxcmrpo+s1V0Rwn/k6lBsFEQxqKDtkUuSt\nIM21+riklZ/T+rhEHW9H/3D7q4V9nG9YUWchxwIDAQABAoIBADs7+l9Ff4FGpYeA\nQD+mBIY5z7MM2Lr30a0/OIofeNRdx9rb01G4A4Nmzkm6ABYk4kKIsgXUKe9BmHJ+\nt6GqnEE49ta1fr/Tjughz1G/r7QSCwhb7jW+lAbGeGt83Q4Ev0HaL6iTBt/+6XYy\ny4yYuFPzw3c9EDMh3JYi9iDgMqZLzdLx3Luh+ChrCx34iDzEZtIcwPkvrcrUwgCY\nG01tRyn0xDmjlDlJKy+++FxTsBXjpZajMMgOprHI9RzdygdcS0kHB2e7FXjlNJIK\nWXXJrtHrdYb16PYEM9ksdwew8tywSsARD+OuzMkjIFcLo6IiCBx55Ab3Zl/+v/wS\n6YdoyjkCgYEA6b6qkZmbDKJqwA7q8f7+D55K4UlB8tWssSNM0avBMlReoijw4Bwq\nBfzlVvnARTJ+mOEAJu7H+cMTuv3DrAPInYJWYTS1PVga4a+AHITiTzbDesSENPKJ\n8i1XRy1BLH2Ql4dNLNYymwP6K2dwKi9MpKDhf83wK8iJY6QKWuZK4rsCgYEAkm6W\nNUYkvoFhxY+4xhMnwWqvwlTdm6tBIg+X4khf3F+W+gp8WdYunFyrBRk32DK/fhFo\nMvDvrbcWWCZ/mHs9Z6Vj6FCh6a7TlMJtUl6nXDVt/LJWklA89zhySf1c/SpDwKIw\nZ61AqnFKodM0TcYvo6BDL6uYsbPo9cT72fHrKmUCgYAoKPYtZfLZzN4CeY2sXl9Q\nV6YN5wRJsKSnGqWMMLJU72IDAn3AQ5aEyNquguKiAPb3VVEtX0FEjFvLeOYxm8TQ\nKyCkOuYY0BZuSDT+sWYgrgwP/8unPTSuQ1QHiqLz/e7l2f1MgtDXCWvmITOIS/Aj\nKPEVgFCPdTkFwZ112LNtGQKBgH5VhqaD0+PAgHgQXVwW3q/Subyxt1g80j7usR17\nT6kzl3A2Z76iOSiSEsKdVT7j2a6MmheJI1/+m/qFjIQjn4CygpXGK0sPmF+5ttPf\n8ght2Gyx+FrXeyHXw1LukrGhKz8hLx3jrezfPYKMU43eNO+4rv6Lz3tROuX/g0Iv\no7K5AoGAOrUD0BvS/BEQ1lmqUmVfvDYMuIzv8VoxecBRPBb4Zw1ojDNAQNNUytwK\nLPDikxj60S5WL2IpN/Db51i/EQCQXLDj52fv0sKdiFiEXbGjhS2kaea6yTkhXyn5\nBxnQYhyVtTTRw8bhbXwePL19dEEfuvRprNXgUOo53yWzORZKh9k=\n-----END RSA PRIVATE KEY-----\n",
    "service": "vcs"
  }
EOF
)
```
- try to start workspace It should fail.

![Знімок екрана 2021-02-10 о 15 37 10](https://user-images.githubusercontent.com/1614429/107517319-47ef0a00-6bb6-11eb-9450-4c760daf1697.png)
- Deploy new che-server image `quay.io/skabashn/che-server:che19024`
- Try to start the workspace again. It should start successfully.
- To delete invalid key use .
```bash
curl  -X DELETE -s $CHE_SERVER_URL'/api/ssh/vcs?name=http%3A%2F%2Fgithub.com'  -H 'Accept: application/json, text/plain'  -H 'Authorization: Bearer '${KEYCLOAK_TOKEN} 
```

### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)


### Reviewers

Reviewers, please comment how you tested the PR when approving it.
